src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs

```diff
@@ -374,69 +253,115 @@ public override void ExecuteSiteRecoveryCmdlet()
                         TfoNicResourceGroupName = this.TfoNicResourceGroupName,
                         TfoReuseExistingNic = this.TfoReuseExistingNic,
-                        TfoVMSubnetName = this.TfoVMSubnetName,
                         TfoNetworkSecurityGroupId = this.TfoNetworkSecurityGroupId,
-                        EnableAcceleratedNetworkingOnTfo = this.EnableAcceleratedNetworkingOnTfo,
-                        TfoIPConfigs =
-                            new List<IPConfig>
-                            {
-                                new IPConfig
-                                {
-                                    StaticIPAddress = this.TfoNicStaticIPAddress,
-                                    PublicIpAddressId = this.TfoPublicIPAddressId,
-                                    LBBackendAddressPoolIds =
-                                        this.TfoLBBackendAddressPoolId?.ToList() ??
-                                        new List<string>()
-                                }
-                            }
+                        EnableAcceleratedNetworkingOnTfo = this.EnableAcceleratedNetworkingOnTfo
                     };
 
                     break;
             }
 
             this.WriteObject(nicConfig);
         }
+
+        // validates IP configs and populate values not provided in input from DB NIC entiry.
+        // returns false in case of validation error.
+        private bool validateAndPopulateIPConfigs(ASRVMNicDetails vmNic)
+        {
+            bool isTfoNetworkRequired = false;
+            bool isRecoveryNetworkRequired = false;
+
+            foreach (var ipConfig in this.IPConfigs)
+            {
+                if (!string.IsNullOrEmpty(ipConfig.TfoSubnetName))
+                {
+                    isTfoNetworkRequired = true;
+                }
+
+                if (!string.IsNullOrEmpty(ipConfig.RecoverySubnetName))
+                {
+                    isRecoveryNetworkRequired = true;
+                }
+
+                var vmNicIPConfig = vmNic.IpConfigs.FirstOrDefault(
+                    ip => ip.Name.Equals(
+                        ipConfig.IpConfigName, StringComparison.OrdinalIgnoreCase));
+
+                if (vmNicIPConfig == null)
+                {
+                    this.WriteWarning(string.Format(Resources.IPConfigNotFoundInVMNic, ipConfig.IpConfigName));
+                    return false;
+                }
+
+                ipConfig.IsPrimary = vmNicIPConfig.IsPrimary;
+                if (string.IsNullOrEmpty(ipConfig.RecoverySubnetName))
+                {
+                    ipConfig.RecoverySubnetName = vmNicIPConfig.RecoverySubnetName;
+                }
+
+                if (ipConfig.RecoveryStaticIPAddress == null)
+                {
+                    ipConfig.RecoveryStaticIPAddress = vmNicIPConfig.RecoveryStaticIPAddress;
+                }
+
+                if (string.IsNullOrEmpty(ipConfig.RecoveryPublicIPAddressId))
```

Comment:
only null check. Update the ValidateIsNotNullOrEmpty to just ValidateIsNotNull

Reply:
updated